crates/opt8n/src/opt8n.rs

```diff
@@ -0,0 +1,303 @@
+                            .storage
+                            .into_iter()
+                            .map(|(k, v)| (B256::from(k), B256::from(v)))
+                            .collect::<BTreeMap<B256, B256>>(),
+                    },
+                )
+            })
+            .collect::<HashMap<Address, AccountState>>();
+
+        Ok(state)
+    }
+
+    /// Updates the pre and post state allocations of the [ExecutionFixture] from Revm.
+    pub fn capture_pre_post_alloc(&mut self, block: &Block) -> Result<()> {
+        let revm_db = AlloyDB::new(
+            self.node_handle.http_provider(),
+            BlockId::from(block.header.number - 1),
+        );
+
+        let block_env = BlockEnv {
+            number: U256::from(block.header.number),
+            coinbase: block.header.beneficiary,
+            timestamp: U256::from(block.header.timestamp),
+            difficulty: block.header.difficulty,
+            gas_limit: U256::from(block.header.gas_limit),
+            prevrandao: Some(block.header.mix_hash),
+            basefee: U256::from(block.header.base_fee_per_gas.unwrap_or_default()),
+            blob_excess_gas_and_price: block
+                .header
+                .excess_blob_gas
+                .map(|excess_gas| BlobExcessGasAndPrice::new(excess_gas as u64)),
+        };
+
+        let mut evm = EvmBuilder::default()
+            .with_ref_db(Box::new(revm_db))
+            .with_block_env(block_env)
+            .build();
+
+        evm.context.evm.env.cfg.chain_id = self.eth_api.chain_id();
+        println!("Transactions length: {}", block.transactions.len());
+        for tx in block.transactions.iter() {
+            let tx_env = to_revm_tx_env(tx.transaction.clone())?;
+            evm.context.evm.env.tx = tx_env;
+            let result = evm.transact()?;
+            println!("{:?}", result);
+            let db = evm.context.evm.db.0.clone();
+            let pre_state_frame = GethTraceBuilder::new(vec![], TracingInspectorConfig::default())
+                .geth_prestate_traces(
+                    &result,
+                    PreStateConfig {
+                        diff_mode: Some(true),
+                    },
+                    db,
+                )?;
+
+            println!("Pre state: {:?}", pre_state_frame);
+
+            if let PreStateFrame::Diff(diff) = pre_state_frame {
+                diff.pre.into_iter().for_each(|(account, state)| {
+                    self.execution_fixture.alloc.entry(account).or_insert(state);
+                });
+                diff.post.into_iter().for_each(|(account, state)| {
+                    self.execution_fixture.out_alloc.insert(account, state);
+                });
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn mine_block(&mut self) {
+        self.eth_api.mine_one().await;
+    }
+
+    pub async fn generate_execution_fixture(&mut self, block: Block) -> Result<()> {
+        self.dump_anvil_state().await?;
```

Comment:
This isn't doing anything if the returned state isn't set on `out_alloc`. My intuition was there's actually 2 conceivable pre/post states. The pre/post state of the entire database before/after the block, and the pre/post state of the Accounts touched by the EVM over all transactions. Maybe we could set `alloc`, `out_alloc` based on a CLI flag (could also be that the latter is the only useful pre/post state to track I'm not sure)

Reply:
Can you elaborate on what you mean here? I am under the impression that we will only need the pre/post state for accounts that have been touched during the execution context rather than all of the state in the db pre execution?